src/main/java/com/binance/dex/api/client/websocket/BinanceDexWSApiImpl.java

```diff
@@ -0,0 +1,161 @@
+import com.binance.dex.api.client.domain.BlockMeta;
+import com.binance.dex.api.client.domain.Proposal;
+import com.binance.dex.api.client.domain.WSMethod;
+import com.binance.dex.api.client.domain.broadcast.Transaction;
+import com.binance.dex.api.client.domain.exception.BinanceDexWSException;
+import com.binance.dex.api.client.domain.jsonrpc.ABCIQueryResult;
+import com.binance.dex.api.client.domain.jsonrpc.BlockInfoResult;
+import com.binance.dex.api.client.domain.jsonrpc.JsonRpcResponse;
+import com.binance.dex.api.client.domain.websocket.AbciQueryRequest;
+import com.binance.dex.api.client.domain.websocket.CommonRequest;
+import com.binance.dex.api.client.domain.websocket.TxSearchRequest;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Maps;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Binance Dex Websocket API implementation.
+ *
+ */
+public class BinanceDexWSApiImpl extends IdGenerator implements BinanceDexWSApi {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    private static final String JSON_RPC_VERSION = "2.0";
+    private BinanceDexClientEndpoint<JsonRpcResponse> endpoint;
+    private TransactionConverter transactionConverter;
+
+    BinanceDexWSApiImpl(BinanceDexEnvironment env,BinanceDexClientEndpoint<JsonRpcResponse> endpoint){
+        this.endpoint = endpoint;
+        this.transactionConverter = new TransactionConverter(env.getHrp());
+    }
+
+    BinanceDexWSApiImpl(String hrp,BinanceDexClientEndpoint<JsonRpcResponse> endpoint){
+        this.endpoint = endpoint;
+        this.transactionConverter = new TransactionConverter(hrp);
+    }
+
+
+    @Override
+    public JsonRpcResponse netInfo() {
+        String id = getId(WSMethod.net_info.name());
+        return endpoint.sendMessage(id,buildWSRequest(WSMethod.net_info.name(),id,null));
+    }
+
+    @Override
+    public BlockMeta.BlockMetaResult blockByHeight(Long height) {
+        String id = getId(WSMethod.block.name());
+        BlockMeta.BlockMetaResult block;
+        Map.Entry params = Maps.immutableEntry("height",String.valueOf(height));
+        String request = buildWSRequest(WSMethod.block.name(),id,params);
+        JsonRpcResponse response = endpoint.sendMessage(id,request);
+        if(response.getError() != null){
+            throw new BinanceDexWSException(id,WSMethod.block.name(),response.getError());
+        }
+        try {
+            block = objectMapper.readValue(objectMapper.writeValueAsBytes(response.getResult()),BlockMeta.BlockMetaResult.class);
+        }catch (IOException e){
+            throw new RuntimeException(e);
+        }
+        return block;
+    }
+
+    @Override
+    public List<Transaction> txSearch(Long height){
```

Comment:
The function name `txSearch` is confusing. Its functionality is to get all transaction at specified height? Am I right? If so, how about changing the name which can imply its functionality.